src/igraph/__init__.py

```diff
@@ -3190,6 +3190,54 @@ def DataFrame(klass, edges, directed=True, vertices=None, use_vids=False):
 
         return g
 
+    def get_vertex_dataframe(self):
```

Comment:
Perhaps we could make it optional to use the `name` attribute as the index or not? There are a few options here. We could allow the user to specify an attribute as index, or simply a boolean indicating or not whether the vertex `name` attribute should be used for creating the dataframe.

Reply:
The order of vertices is by id, not name, so one can currently index the vertex df by id or name as usual:

```python
df.iloc[myid]
df.loc[myname]
```

Most variations are one-liners, e.g.:

```python
df.index = np.arange(df.shape[0])  # From names to ids
df.set_index(mycol, inplace=True)  # From name/id to another column
```
so it's not hard for the user to achieve exactly what they want.

Instead, I would establish a decently consistent convention between nodes and edges, e.g.

1. using vertex ids as index, then in the source/target of the edge df we use vertex ids (this is my favourite)
2. using vertex names as index (current), then the source/target of the edge df use vertex names (less good because if the vertices have no names we get integers, otherwise strings, which tends to become a mess).

A bunch of ifs are not a problem, but I feel like telling the user about these one-liners in the docstring is worth more than baby-feeding them the exact variation.